src/tests/data/tls-policy/bsi.txt

```diff
@@ -10,7 +10,7 @@ signature_hashes = SHA-512 SHA-384 SHA-256
 macs = AEAD SHA-384 SHA-256
 key_exchange_methods = ECDH DH ECDHE_PSK
 signature_methods = ECDSA RSA DSA
-key_exchange_groups = brainpool512r1 brainpool384r1 brainpool256r1 secp521r1 secp384r1 secp256r1 ffdhe/ietf/4096 ffdhe/ietf/3072
```

Comment:
IMHO the policy files should not be touched by this change. Users may find it confusing that brainpool curves are present twice here each, I think it would become a common source for error. "brainpool512r1" should be matched by the code to the corresponding code point automatically, just as `Group_Params::to_algorithm_spec()` does the other way around.

Reply:
I fully agree on the potential end user confusion. Though, I'm somewhat on the fence whether users may still need the additional flexibility in configuration that it brings.

_Just to make sure we're on the same page:_ You're suggesting, that `brainpool384r1` (for instance) should translate into advertising **both** code points (and potentially offering key share values for both in a TLS 1.3 ClientHello), right?

On the one hand, this would somewhat complicate the logic of the code point selection and introduce a potential source of bugs (as we will likely need to handle this Brainpool-specific special case in more than one place). On the other hand, it would prevent users from explicitly choosing one code point over the other.

This is explicitly not meant to be an objection, though. However, if we introduce the additional logic, I want to suggest a compromise: Let's add the following configuration options:

* `brainpool*r1` (the existing name) - add both code points
* `brainpool*r1_legacy` - add the old code points only
* `brainpool*r1_tls13` - add the new code points only

That approach would also make the "special case" more explicit in the code base. The existing name could map to some special `Group_Params_Code::BRAINPOOL_META_VALUE` which we can handle explicitly in a special case (and map to `::BRAINPOOL_TLS12` and `::BRAINPOOL_TLS13`).

Whatever we do: its a mess. 😢